### ISSUE TITLE
covid19_cases_map.ipynb: Problem in installation of cartopy

### ISSUE THREAD

**Issue by lxlofpku** [stefmolin/Hands-On-Data-Analysis-with-Pandas-2nd-edition#30]
### Required attestation
- [Y] I have **completely and exactly** followed the virtual environment setup instructions from the book and have consulted the [README](https://github.com/stefmolin/Hands-On-Data-Analysis-with-Pandas-2nd-edition#notes-on-environment-setup) for any updates.
- [Y] I have cloned either this repository or my fork of this repository so that I have all necessary files on my local machine.
- [Y] I have checked that this issue has not already been reported or resolved [here](https://github.com/stefmolin/Hands-On-Data-Analysis-with-Pandas-2nd-edition/issues?q=is%3Aissue+label%3Atroubleshooting).
- [Y] I am aware that there is a [pre-built Binder environment](https://mybinder.org/v2/gh/stefmolin/binder-environments/2nd_edition?urlpath=git-pull%3Frepo%3Dhttps%253A%252F%252Fgithub.com%252Fstefmolin%252FHands-On-Data-Analysis-with-Pandas-2nd-edition%26urlpath%3Dlab%252Ftree%252FHands-On-Data-Analysis-with-Pandas-2nd-edition%252F%26branch%3Dmaster) that I can use, but I want to install locally on my machine instead.

---

##### 1. Which OS are you using?
TODO: Provide your OS here &ndash; make sure to differentiate between Intel and M1 chip Macs.

Mac M2.

##### 2. Which Python version are you using?
TODO: Provide your Python version here.

python 3.9 for base and python 3.8 for book_env

##### 3. Are you using `conda` or `venv`?
TODO: Indicate whether you are using `conda` or `venv`.

conda

---

### Commands run and their outputs
Please provide **all** of the commands you ran as well as the traceback:

Dear Stefanie:

I have some problems in running the "covid19_cases_mpa.ipynb"  in chapter 6 and installation of cartopy. The detail descriotion is in the "details" section.

Any help will be highly appreciated.

Best regards,
Li-Xin Li

(Before posting this message I did a search on google but did not get a clear solution.)

<details>

First, I installed cartopy with "base" by running "conda install -c conda-forge cartopy". I found that I cannot import cartopy as I run the jupyter lab in "book_env". As I run "import cartopy.crs as ccrs" in the jupyter notebook "covid19_cases_map.ipynb" I got a message saying that "ModuleNotFoundError: no module named 'cartopy'".

Then, in a terminal with "book_env" environment, I run "conda install -c conda-forge cartopy" again (without uninstallation of the cartcopy in "base" environment). Then in a jupyter notebook in "book_env" I can run "import cartopy.crs as ccrs" with no problem. That is, I went through "covid19_cases_map.ipynb" with out a problem.

However, after I finished "covid_cases_map.ipynb" then started dong the exercises in chapter 6 in a new jupyter notebook, I got another problem:

In a jupyter notebook in "boon_env" I run 
#---------------------------------------------------------------------
%matplotlib inline
import matplotlib.pyplot as plt
import numpy as np
import pandas as pd
import seaborn as sns

fb = pd.read_csv('data/fb_stock_prices_2018.csv', index_col='date', parse_dates=True)
quakes = pd.read_csv('data/earthquakes.csv')
covid = pd.read_csv('data/covid19_cases.csv').assign(
    date=lambda x: pd.to_datetime(x.dateRep, format='%d/%m/%Y')
).set_index('date').replace(
    'United_States_of_America', 'USA'
).sort_index()['2020-01-18':'2020-09-18']
#------------------------------------------------------------------

I got the following mesage:

"\<frozen importlib._bootstrap\>:219: RuntimeWarning: scipy._lib.messagestream.MessageStream size changed, may indicate binary incompatibility. Expected 56 from C header, got 64 from PyObject"

I never had this problem before installing cartopy.

Does my process in installing cartopy has some problems? By installing a cartopy in "base" then installing another cartopy in "book_env", I realize that there may be two different versions of cartopy on my M2. Does this cause some problem of inconsistency? Should I uninstall the one with the "base" enviroment? 

I admit that after installing the cartopy, I updated the "anacondo" by running "conda update -n base -c defaults conda" as being reminded during installation of cartopy. Did this case the problem?

```
TODO: paste commands and any traceback here
```

</details>

### Screenshots
Optionally, include any screenshots that will help diagnose the issue.
![Xnip2023-01-16_15-36-02](https://user-images.githubusercontent.com/115785359/212622629-46a0c1e0-0b5d-44be-9b3e-c2e4025c8b85.jpg)

**Comment by stefmolin** [comment]
You shouldn't install things in the base environment in general. Based on your error, I believe your version of scipy changed. Does your setup still pass the checks from chapter 1 (checking_your_setup.ipynb)?

If anything has changed such that it no longer passes the checks, you will need to downgrade those packages or recreate the environment as you did in chapter 1.

**Comment by lxlofpku** [comment]
Hi Stefanie:

I reinstalled everthing, but got the same problem. I reinstalled anaconda3 (the same version as before), created an environment book_env with Python 3.8, then reinstalled required packages for your book (pip3 install -r requirements.txt), then installed cartopy (run 'conda install -c conda-forge cartopy' in book_env). Then I still get the message
"\<frozen importlib._bootstrap\>:219: RuntimeWarning: scipy._lib.messagestream.MessageStream size changed, may indicate binary incompatibility. Expected 56 from C header, got 64 from PyObject"
exactly as before.

While for checking up the set up, I got the result as shown in the figure below so it didn't go through. But this problem existed before. Since it did not affect my working with other notebooks (text and excercises) in your book, I did not report this problem to you. But of course if you can give me somme highlight I will be very grateful.

Best wishes,
Li-Xin

![Xnip2023-01-17_22-05-00](https://user-images.githubusercontent.com/115785359/212926128-4cdf5c2e-b8f7-49ee-b3bf-a837c988001f.jpg)

**Comment by lxlofpku** [comment]
@stefmolin 

Update:

I reinstalled anaconda3 with the latest version. Then I create book_env with Python 3.8 (as before). Then, I installed the required python packages for the book as described in chapter 1, in the book_env environment. 

Then, I run checking_your_setup.ipynb to check the set up. It passed the check without a problem. Everythong is OK so far.

Then, I installed cartopy by 'conda install -c conda-forge cartopy' in book_env. The installation process went through smoothly. With "conda list" I see that the scipy has the version 1.7.3, both before and after the installation of cartopy. That is, as I run conda list in book_env, I got
..........................
scipy                     1.7.3                    pypi_0    pypi
...........................
both before and after the installation of cartopy). 
(But during the installation process I saw the following message:
"
The following NEW packages will be INSTALLED:
...........................
scipy              conda-forge/osx-64::scipy-1.10.0-py38hfb8b963_0 None
...........................
"
and after the installation I did not see scipy-1.10.0.)

Then I run checking_your_setup.ipynb to check the set up again. This time the check did not get through. I stll get the message
\<frozen importlib._bootstrap\>:219: RuntimeWarning: scipy._lib.messagestream.MessageStream size changed, may indicate binary incompatibility. Expected 56 from C header, got 64 from PyObject

The details are in the follwoing figures.
![check1](https://user-images.githubusercontent.com/115785359/213091722-513366a3-70d2-48af-96a7-278fa1b1bbff.jpg)
![check2](https://user-images.githubusercontent.com/115785359/213091757-c76fc5f5-6365-497c-bfa1-f269765230a0.jpg)
![check3](https://user-images.githubusercontent.com/115785359/213091770-433e0b6a-dd66-454f-8467-19f4a55146c4.jpg)
![check4](https://user-images.githubusercontent.com/115785359/213091785-327d9c44-6953-4351-b7f7-56ec0c33b8cc.jpg)

Any help will be great appreciated!

Best wishes,
Li-Xin

**Comment by lxlofpku** [comment]
Update again:

I think the problem is solved. By running 'scipy.__version__' and 'np.__version__' in the notebook I found that the versions of scipy and numpy had indeed been changed by the installation of cartopy. (I still don't understand why the command 'conda list' in the terminal does not show the change.) After downgrading the versions of scipy and numpy to the versions before installation of cartopy, the problem mentioned in the above post disappears. However, as I ran 'checking_your_setup.ipynb to check' I got two messages saying that the versions of matlibplot and requests also need be adjusted. After installlation of matlibplot and requests with the required versions (a little bit complicated since I had to find the channels for downloading the required packages), everything seems OK now.

Thank you very much for your kind help! (Now I know that I should not install packages in the base environment in general.)

Best wishes,
Li-Xin Li
